src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java

```diff
@@ -1,27 +1,181 @@
+import uk.gov.companieshouse.presenter.account.model.request.presenter.account.PresenterName;
+import uk.gov.companieshouse.presenter.account.model.request.presenter.account.PresenterRequest;
+import uk.gov.companieshouse.presenter.account.service.PresenterAccountService;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.matchesPattern;
+
+@ExtendWith(MockitoExtension.class)
+class PresenterAccountControllerTest {
+
+    private static final String PRESENTER_ACCOUNT_REGEX = "/presenter-account/[0-9a-z]{1,100}";
+    private static final String PRENETER_ACCOUNT_NULL = "/preneter-account/null";
+    private static final String PRESENTER_ACCOUNT = "/presenter-account/";
+    private final static String USER_ID = "userId";
+    private final static String EMAIL = "test@test.test";
+    private final static String FIRST_NAME = "test";
+    private final static String LAST_NAME = "last";
+    private final static String PREMISES = "premise";
+    private final static String FIRST_LINE = "first line";
+    private final static String SECOND_LINE = "second line";
+    private final static String COUNTY = "county";
+    private final static String COUNTRY = "country";
+    private final static String POSTCODE = "postcode";
 
-public class PresenterAccountControllerTest {
 
     PresenterAccountController presenterAccountController;
 
+    @Mock
+    PresenterAccountService presenterAccountService;
+
+    @Mock
+    Logger logger;
+
     @BeforeEach
     void setUp() {
-        presenterAccountController = new PresenterAccountController();
+        presenterAccountController = new PresenterAccountController(presenterAccountService, logger);
     }
 
     @Test
     @DisplayName("Return 200 on successfully health check")
-    void test_HealthCheck_Endpoint_for_SuccessResponse (){
+    void testHealthCheckEndpointSuccessResponse (){
         var response = presenterAccountController.healthCheck();
         assertThat(response.getStatusCode(), is(HttpStatus.OK));
     }
 
+    @Test
+    @DisplayName("Return 201 when valid presenter details is submitted")
+    void testCreatePresenterAccountSuccessResponse() {
+        var presenterDetails = createPresenterDetails(
+            USER_ID,
+            EMAIL,
+            FIRST_NAME,
+            LAST_NAME,
+            PREMISES,
+            FIRST_LINE,
+            SECOND_LINE,
+            COUNTY,
+            COUNTRY,
+            POSTCODE);
+        var response = presenterAccountController.createPresenterAccount(presenterDetails);
+        var header = response.getHeaders().getFirst("Location");
+        assertThat(response.getStatusCode(), is(HttpStatus.CREATED));
+        assertTrue(header.contains(PRESENTER_ACCOUNT));
+        assertFalse(header.contains(PRENETER_ACCOUNT_NULL));
+        assertThat(matchesPattern(PRESENTER_ACCOUNT_REGEX).matches(header), is(true));
```

Comment:
You can mock the return of the presenter account service so it returns a specific id, and then check that, as opposed to using a regex